### PR TITLE
editoast: move sql request from `views::infra` to `modelsv2::infra`

### PR DESCRIPTION
**Pull request by hamz2a**
Part of https://github.com/OpenRailAssociation/osrd/issues/6397

- [x] views::infra::railjson
- [x] views::infra::routes
- [x] views::infra::edition

### CODE REVIEW

File: editoast/src/modelsv2/infra.rs [hunk 1]
```diff
@@ -257,6 +262,30 @@ impl Infra {
...
+    ) -> Result<Vec<SplitedTrackSectionWithData>> {
+        let query = format!("SELECT
+            object_table.obj_id as obj_id,
+            object_table.data as railjson,
+            ST_AsGeoJSON(ST_LineSubstring(ST_GeomFromGeoJSON(object_table.data->'geo'), 0, $3))::jsonb as left_geo,
+	        ST_AsGeoJSON(ST_LineSubstring(ST_GeomFromGeoJSON(object_table.data->'geo'), $3, 1))::jsonb as right_geo
+            FROM {} AS object_table
+            WHERE object_table.infra_id = $1 AND object_table.obj_id = $2",
+            get_table(&ObjectType::TrackSection),
+        );
```
Comment (woshilapin) [inline editoast/src/modelsv2/infra.rs]: If we want to continue on this path of homogeneisation, then this SQL request can be put in a separate file.

**Review by woshilapin — APPROVED** [review]
And a bunch more 🎉

File: editoast/src/modelsv2/infra.rs (hunk 1 above)
Comment (hamz2a) [reply editoast/src/modelsv2/infra.rs]: this one is a bit unusual, we have the name of the table which is a variable, and I don't know how to put it as a parameter in `sql_query`

File: editoast/src/modelsv2/infra/splited_track_section_with_data.rs
Comment (leovalais) [inline editoast/src/modelsv2/infra/splited_track_section_with_data.rs:1]: I'm a bit skeptical about putting every function in `infra.rs` and placing the deserializer in separate files. My suggestion would be to move `fn get_splited_track_section_with_data` in an `impl Infra` block in this file. 

@woshilapin, I know you find this practice odd but I believe that would improve organization (and de-clutter a bit `infra.rs`). Wdyt?

File: editoast/src/modelsv2/infra/splited_track_section_with_data.rs [hunk 2]
```diff
@@ -0,0 +1,17 @@
+use diesel::sql_types::Jsonb;
+use diesel::sql_types::Text;
+use editoast_schemas::infra::TrackSection;
+use serde::Deserialize;
+use serde::Serialize;
+
+#[derive(QueryableByName, Debug, Clone, Serialize, Deserialize)]
```
Comment (leovalais) [inline editoast/src/modelsv2/infra/splited_track_section_with_data.rs]: I'm not sure that `Serialize` is really needed.

File: editoast/src/modelsv2/infra.rs [hunk 3]
```diff
@@ -303,6 +305,19 @@ impl Infra {
             .await?;
         Ok(routes)
     }
+
+    pub async fn get_railjson(
+        conn: &mut DbConnection,
+        infra_id: i64,
+        table_name: &str,
```
Comment (leovalais) [inline editoast/src/modelsv2/infra.rs]:
```suggestion
        object_type: &ObjectType,
```

File: editoast/src/modelsv2/infra.rs (hunk 1 above)
Comment (leovalais) [reply editoast/src/modelsv2/infra.rs]: Yeah I recall trying to do that a long time ago and I believe that's impossible. Ideally to avoid that we'd rewrite the request using diesel's DSL, but since there's postgis functions and all that's not trivial.

However since the table is constant here I'd say it's fine to directly write `infra_object-tracksection`

File: editoast/src/modelsv2/infra.rs [hunk 4]
```diff
@@ -257,6 +266,58 @@ impl Infra {
             .await?;
         Ok(speed_limits_tags)
     }
+
+    pub async fn get_splited_track_section_with_data(
+        &self,
+        conn: &mut DbConnection,
+        track: Identifier,
+        distance_fraction: f64,
+    ) -> Result<Vec<SplitedTrackSectionWithData>> {
```
Comment (leovalais) [inline editoast/src/modelsv2/infra.rs]:
```suggestion
    ) -> Result<Option<SplitedTrackSectionWithData>> {
```

The `Vec` is not required here since we'll have at most one object (if we have zero it means that the cache is outdated, but it's up to the parent function to panic or handle the error).

**Review by leovalais — CHANGES_REQUESTED** [review]
Thanks for the cleanup! :) I have a few comments, partly about problems that were already there, but since you're refactoring this part we might as well tackle them now.

File: editoast/src/modelsv2/infra.rs (hunk 4 above)
Comment (hamz2a) [reply editoast/src/modelsv2/infra.rs]: Done.

File: editoast/src/modelsv2/infra/splited_track_section_with_data.rs [hunk 5]
```diff
@@ -0,0 +1,16 @@
+use diesel::sql_types::Jsonb;
+use diesel::sql_types::Text;
+use editoast_schemas::infra::TrackSection;
+use serde::Deserialize;
+
+#[derive(QueryableByName, Debug, Clone, Deserialize)]
+pub struct SplitedTrackSectionWithData {
```
Comment (woshilapin) [inline editoast/src/modelsv2/infra/splited_track_section_with_data.rs:17]: I wonder if we should rename this (and all functions and SQL migration, etc.) because `splited` is not really a word https://english.stackexchange.com/questions/467873/the-past-participle-of-split-split-or-splitted#467898. This might be ok to do in another PR if the renaming is too wide.

File: editoast/src/modelsv2/infra/splited_track_section_with_data.rs
Comment (woshilapin) [reply editoast/src/modelsv2/infra/splited_track_section_with_data.rs:1]: What if we change our point of view. What if we expose the following function instead?

```rust
impl SplitedTrackSectionWithData {
  fn get_with_data(
    conn: &mut DbConnection,
    infra_id: i64,    
    track: Identifier,
    distance_fraction: f64,)
{
  unimplemented!()
}
```

Comment (hamz2a) [reply editoast/src/modelsv2/infra/splited_track_section_with_data.rs:1]: I prefer to keep everything in `infra.rs`, I don't think that the infra is as long as this, a little almost 200 lines for the `impl` part.

Comment (leovalais) [reply editoast/src/modelsv2/infra/splited_track_section_with_data.rs:1]: I disagree, the file is more than 500 lines including the tests. That's too much. A proof of that is that you put the `QueryableByName` structs in other files. Either (1) we're okay with a big file in which case we don't need the `infra/` folder (we're not in Java we don't need to create a new file for each struct), or (2) we split the `Infra` interface into several files, but then we also need to offload the `Infra` implementation.

---

Well if we're okay with changing that much the interface I'd be more in favor of:

```rust
// in infra_objects.rs
impl TrackSectionModel {
    fn split(conn: _, distance_fraction: f64) -> Result<(schema::TrackSection, schema::TrackSection)>;
}
```

This way the (infra_id, obj_id) is already known, we don't need to expose the DB result type and that would simplify a bit the endpoint that's too big anyway.

---

On a more general note, this comment also applies to all other `QueryableByNames` moved. I believe it's more natural to write:

```rust
infra.get_routes_from_waypoint(conn, id, type)
```

rather than:

```rust
RoutesFromWaypointsResult::get(&infra, conn, id, type)
```

(But it's not a hill I'm willing to die on either...)

My point is that I don't want to overcrowd a single file, but that doesn't prevent us from keeping `Infra` as the main interface to interact with infras.

---

Not sure how to proceed, do we merge this PR as is and changes come later? @woshilapin

Comment (woshilapin) [reply editoast/src/modelsv2/infra/splited_track_section_with_data.rs:1]: We discussed it with @leovalais and he almost convinced me. So whichever solution you keep is fine.

Comment (hamz2a) [reply editoast/src/modelsv2/infra/splited_track_section_with_data.rs:1]: Ok, I will move each function into the appropriate file.

**Review by leovalais — APPROVED** [review]
That's great, lgtm, thanks! 🚀
